### PR TITLE
feat: support file_size config in file and image upload

### PR DESCRIPTION
**Pull request by khushal87**
## 🎯 Goal

The goal of the PR is to support `file_size` image and file config such the file size exceeding the limit cannot be uploaded. The limit can be controlled or set from the dashboard. The default is 100 MB in the SDK. 

There are a couple of improvements to how app settings are fetched on mounted and improving the code of existing file config.

## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [x] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

File: package/src/contexts/messageInputContext/MessageInputContext.tsx [hunk 1]
```diff
@@ -1191,18 +1183,25 @@ export const MessageInputProvider = <
...
-      (mimeType: string) => file.name?.includes(mimeType),
-    );
+    const isAllowed = isUploadAllowed({ config: fileConfig, file });
+
+    const sizeLimit = size_limit || MAX_FILE_SIZE_TO_UPLOAD_IN_MB;
+
+    if (file.size && file?.size > sizeLimit) {
+      Alert.alert(
+        t('File is too large: {{ size }}, maximum upload size is {{ limit }}', {
+          limit: sizeLimit / (1024 * 1024),
```
Comment (santhoshvai) [inline package/src/contexts/messageInputContext/MessageInputContext.tsx:1196]: can we also do prettifyFileSize in the limit

File: package/src/contexts/messageInputContext/MessageInputContext.tsx [hunk 2]
```diff
@@ -1220,26 +1219,33 @@ export const MessageInputProvider = <
...
-    const isBlockedImageExtension = blockedImageExtensionTypes?.some((imageExtensionType: string) =>
-      image.uri?.includes(imageExtensionType),
-    );
+    const isAllowed = isUploadAllowed({ config: imageUploadConfig, file: image });
+
+    const sizeLimit = size_limit || MAX_FILE_SIZE_TO_UPLOAD_IN_MB;
+
+    if (image.size && image?.size > sizeLimit) {
+      Alert.alert(
+        t('File is too large: {{ size }}, maximum upload size is {{ limit }}', {
```
Comment (santhoshvai) [inline package/src/contexts/messageInputContext/MessageInputContext.tsx:1239]:
```suggestion
        t('Image is too large: {{ size }}, maximum upload size is {{ limit }}', {
```

File: package/src/contexts/messageInputContext/MessageInputContext.tsx [hunk 3]
```diff
@@ -1191,18 +1183,25 @@ export const MessageInputProvider = <
...
-      (fileExtensionType: string) => file.name?.includes(fileExtensionType),
-    );
-    const isBlockedFileMimeType: boolean | undefined = blockedFileMimeTypes?.some(
-      (mimeType: string) => file.name?.includes(mimeType),
-    );
+    const isAllowed = isUploadAllowed({ config: fileConfig, file });
+
+    const sizeLimit = size_limit || MAX_FILE_SIZE_TO_UPLOAD_IN_MB;
+
+    if (file.size && file?.size > sizeLimit) {
```
[the remark this answers is not in the document]
Comment (khushal87) [reply package/src/contexts/messageInputContext/MessageInputContext.tsx:1193]: Ah, no, I missed changing the var name. Let me quickly do that. Thanks for pointing it out @myandrienko 😄

**Comment by khushal87** [comment]
All the suggestions are added @myandrienko @santhoshvai. Thanks 😄

File: package/src/contexts/messageInputContext/MessageInputContext.tsx [hunk 4]
```diff
@@ -1191,18 +1182,26 @@ export const MessageInputProvider = <
...
+    const sizeLimit = size_limit || MAX_FILE_SIZE_TO_UPLOAD;
+
+    if (file.size && file.size > sizeLimit) {
+      Alert.alert(
+        t('File is too large: {{ size }}, maximum upload size is {{ limit }}', {
+          limit: prettifyFileSize(sizeLimit),
+          size: prettifyFileSize(file.size),
+        }),
+      );
+      console.log(selectedFiles);
```
Comment (santhoshvai) [inline package/src/contexts/messageInputContext/MessageInputContext.tsx:1199]:
```suggestion
```

**Review by santhoshvai — APPROVED** [review]

**Pull request merged**